packages/ui-extensions/src/shared.ts

```diff
@@ -11,3 +11,76 @@ export type ComponentsBuilder<ComponentTypes> = {
 
 export type AnyComponentBuilder<ComponentTypes> =
   ComponentsBuilder<ComponentTypes>[keyof ComponentsBuilder<ComponentTypes>];
+
```

Comment:
cc/ @vividviolet I moved `I18n` here. WDYT?

Reply:
Can you put it in `ui-extensions/src/api.ts` instead since it's an api?